### PR TITLE
Prevent IntroView Present Loop

### PR DESCRIPTION
**Pull request by evancrow**
Because of the weird delegate layout in `TabManager` which needs to be refactored, there were multiple calls made to `presentIntroViewController` from `decidePolicyFor navigationAction`. Now before presenting the `IntroView`, check to make sure a `FullScreenModal` isn't already presented.

### How was this tested?
- [x] I tested this manually

## Issues addressed
Fixed: #3705

### CODE REVIEW

File: Client/Frontend/Browser/BrowserViewController/BrowserViewController+IntroView.swift [hunk 1]
```diff
@@ -17,6 +17,11 @@ extension BrowserViewController {
         completion: (() -> Void)? = nil,
         onDismiss: (() -> Void)? = nil
     ) {
+        // Confirm a FullScreenView isn't already visible.
+        if case .fullScreenModal = overlayManager.currentOverlay {
```
Comment (chung-lai) [inline Client/Frontend/Browser/BrowserViewController/BrowserViewController+IntroView.swift:21]: Pretty sure this cover most cases. Wonder whether we can also check the current full screen modal is indeed a login modal?

Comment (chung-lai) [reply Client/Frontend/Browser/BrowserViewController/BrowserViewController+IntroView.swift:21]: Sure, i think that's sufficient too.

**Review by chung-lai — APPROVED** [review]
LGTM. One thing worth confirming if `dismiss` is being called all the time even on iPad when user taps on the background to dismiss the screen.

**Comment by chung-lai** [comment]
May worth cherry picking this.

**Comment by evancrow** [comment]
> LGTM. One thing worth confirming if `dismiss` is being called all the time even on iPad when user taps on the background to dismiss the screen.

Good catch! Moved the code so it will be called for all interaction paths.

**Pull request merged**